scripts/contrib-patch-tests.py

```diff
@@ -0,0 +1,58 @@
+import ast
```

Comment:
Can we add a docstring here for what and how this script works?

- What does it do?
- How do we run it?
- Do we need to delete and regenerate the files if the template changes?

Reply:
> Do we need to delete and regenerate the files if the template changes?

I was a bit in two minds about this. For this draft I decided to generate `test_{contrib}_patch.py`. This means we won't be able to tell (easily) what was generated and what was already there. Perhaps the better idea is to use `test_{contrib}_patch_generated.py`? Or add a special comment at the top (`# This is autogenerated. Do not remove this header if ....`). WDYT?

Reply:
Yeah I love the the `..._generated.py` and the `# This is autogenerated. Do not remove this header if ....` ideas!